Automattic/jetpack
# Fixes CSS RTL handling.

#### Changes proposed in this Pull Request:
* Synchronizes RTL styles for Recipe shortcodes with dotcom by removing the RTL file and relying on auto-generated one.
* Fixes eslint errors in `gulpfile`.
* Enables advanced JavaScript features in `gulpfile` via Babel.
* Starts moving out gulp tasks to separate files to simplify and make `gulpfile` more DRY.

#### Testing instructions:
* Make sure you can `yarn build` successfully, even with `env NODE_ENV=production`.
* Enable shortcodes, create a post with a recipe shortcode: `[recipe title="Something tasty." time="400 minutes"]Some tasty recipe.[/recipe]`
* Make sure the shortcode uses RTL styles when you switch your site to an RTL language like Arabic or Hebrew.

Note: there's a lot of changes, but almost all `gulpfile` changes are cosmetic.
